### PR TITLE
Add Brave Ads status headers to search.brave.com

### LINKED ISSUE #6024: Add Brave Ads status headers to search.brave.com
### Description: 
In preparation for Brave Ads on Brave Search, it is helpful for the Brave Search backend to understand whether the requesting browser has Brave Ads enabled. As such, we should add a header to any call to Brave Search (and its associated dev/staging environments), whether originating from the main search box or the Omnibox: `X-Brave-Ads-Enabled` should be set to `1` when Brave Ads is enabled on the device.

This issue may provide some guidance on the best way to accomplish that: https://github.com/brave/brave-ios/issues/887

### PR DESCRIPTION
**Pull request by tmancey**
## Summary of Changes

This pull request fixes [#6024](https://github.com/brave/brave-ios/issues/6024)

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

Confirm header is sent if Rewards/ads are enabled
Confirm header is not sent if Rewards/ads are not enable
Confirm header is not sent for non Brave Search URLs

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — CHANGES_REQUESTED** [review]
Let's have a security review for it

**Comment by tmancey** [comment]
@iccub https://github.com/brave/security/issues/1035, thanks

**Review by iccub — APPROVED** [review]
lgtm, let's wait for Brandon's review too

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: Should it be appended in private browsing mode?

Comment (tmancey) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: @kylehickinson ads should not work in Private browsing mode, can you please confirm? Thanks

Comment (stoletheminerals) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: Right now it sends `x-brave-ads-enabled` when in private mode, I think we need to implement an additional check here

Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: they don't but the `isEnabled` property does not check for pbm I assume, Kyle to confirm.
Extra safety never hurts imo

Comment (tmancey) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: @stoletheminerals fixed. Thanks

**Review by stoletheminerals — APPROVED** [review]

**Review by iccub — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift
Comment (remusao) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift:330]: Isn't this gonna prevent the logic below related to Google fallback? In particular, aren't we relying on `tab?.injectResults()` being called in all cases to un-hide the results?

Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift:330]: It should get there at the second loop. First loop adds the header, then second loop handles the fallback.
On iOS we have to cancel and reload request in order to add headers, we can test it when new builds are uploaded

**Pull request merged**
